test/common/src/main/kotlin/org/codice/compliance/verification/core/CoreVerifier.kt

```diff
@@ -20,12 +20,59 @@ import org.codice.compliance.SAMLCore_5_4_2_a
 import org.codice.compliance.SAMLCore_5_4_2_b
 import org.codice.compliance.SAMLCore_5_4_2_b1
 import org.codice.compliance.SAMLCore_6_1_b
+import org.codice.compliance.SAMLSpecRefMessage
 import org.codice.compliance.allChildren
 import org.codice.compliance.children
 import org.codice.compliance.utils.TestCommon.Companion.ELEMENT
+import org.codice.compliance.utils.TestCommon.Companion.REQUESTER
+import org.codice.compliance.utils.TestCommon.Companion.VERSION_MISMATCH
 import org.w3c.dom.Node
 
 class CoreVerifier(val node: Node) {
+
+    /**
+     * Verifies that a response has a status code of urn:oasis:names:tc:SAML:2.0:status:Requester.
+     * This should be called explicitly if an error of this type is expected.
+     */
+    fun verifyRequesterError(samlErrorCode: SAMLSpecRefMessage) {
+        verifyAndReturnStatus(samlErrorCode).children("Status Code").forEach {
+            val code = it.attributes.getNamedItem("Value").textContent
+            if (code != REQUESTER)
+                throw SAMLComplianceException.createWithPropertyMessage(samlErrorCode,
+                        property = "Status Code",
+                        actual = code,
+                        expected = REQUESTER,
+                        node = node)
+        }
+    }
+
+    /**
+     * Verifies that a response has a status code of urn:oasis:names:tc:SAML:2.0:status:VersionMismatch.
+     * This should be called explicitly if an error of this type is expected.
+     */
+    fun verifyVersionMismatchError(samlErrorCode: SAMLSpecRefMessage) {
+        verifyAndReturnStatus(samlErrorCode).children("Status Code").forEach {
+            val code = it.attributes.getNamedItem("Value").textContent
+            if (code != VERSION_MISMATCH)
+                throw SAMLComplianceException.createWithPropertyMessage(samlErrorCode,
+                        property = "Status Code",
+                        actual = code,
+                        expected = REQUESTER,
```

Comment:
❗️ This should say `VERSION_MISMATCH` instead of `REQUESTER`. Also, couldn't you put all of this logic into the common private method below and pass in the URI you're expecting? i.e. `VERSION_MISMATCH` or `REQUESTER`.

Reply:
Oops